File: backend/src/main/kotlin/com/sorsix/eventmanager/domain/Event.kt
package com.sorsix.eventmanager.domain

import com.sorsix.eventmanager.domain.user.User
import jakarta.persistence.*
import java.time.LocalDate

import java.time.LocalDateTime
import java.time.LocalTime

@Entity
@Table(name = "events")
data class Event(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long,

    @Column(length = 100)
    var name: String,

    @Column(length = 1500)
    var description: String,

    var availableTickets: Int,

    var longitude: String,
    var latitude: String,

    @Enumerated(EnumType.STRING)
    var category: Category,

    @ManyToMany
    var tags: MutableList<Tag>,

    var dateStart: LocalDate,

    var dateFinish: LocalDate,

    var timeStart: LocalTime,

    var timeFinish: LocalTime,

    var meetingUrl: String,

    var type: String,

    var price: Double,

    @ManyToOne
    val creator: User,

//    @OneToMany(mappedBy = "event", cascade = [CascadeType.ALL], orphanRemoval = true)
//    val images: MutableList<Image>

//    var thumbnailUrl: String

//    @OneToMany
//    var images: List<Image>
)

File: frontend/src/app/interfaces/event.ts
import { Time } from "@angular/common";
import { User } from "./user";

export interface EventInterface {
    id: number;
    name: string;
    description: string;
    availableTickets: number;
    longitude: string;
    latitude: string;
    category: string; // moze da e problem deka e ENUM
    tags: Tag[];
    dateStart: Date;
    dateFinish: Date;
    timeStart: Time;
    timeFinish: Time;
    creator: User;
    meetingUrl: string,
    type: string,
    price: number,
}

// enum Category {
//     Define your category options here
// }

export interface Tag {
    id?: number;
    name: string;
}
